File: Java/OperatorsPart4.java
//Java program implementing Ternary and Instanceof operator

class OperatorsPart4
{
public static void main(String args[])
{
int age = 20;
String result = (age >= 18) ? "Adult" : "Minor";
System.out.println("Ternary Operator"); /*Ternary operator is the short form of if-else. (condition) ? true : false;*/
System.out.println(result);

System.out.println("\nInstance Of Operator"); /*Used to check whether the object belongs to the class or not. returns true or false*/
class1 c1 = new class1();
System.out.println("Object c1 belongs to class1: " + (c1 instanceof class1));
System.out.println("Object c1 belongs to class2: " + (c1 instanceof class2));

class2 c2 = new class2();
System.out.println("Object c2 belongs to class1: " + (c2 instanceof class1));
System.out.println("Object c2 belongs to class2: " + (c2 instanceof class2));
//since class2 extends class1 object c2 belongs to both of them
}
}
class class1{}
class class2 extends class1{}

File: Kotlin/OperatorsPart4.kt
//Kotlin program implementing Range, Safe call, Elvis operator

fun main()
{
println("Range Operator")
println("Open range") //Closed range. includes the end value
for(i in 1..5){ println("$i") }
println("\nHalf op range") //Half open range. excludes the end value
for(i in 1..<5){ println("$i") }

var name: String? = null /* ? can be used while declaring variable to give null value [var name: String = null] gives error*/
println("\nSafe call operator")
println(name?.length) /*Safe call is used to call a function or access a variable only if it is not a null value, else it returns null without crashing the program*/
println("\nElvis operator")
println(name ?: "Empty") /*Elvis operator is used to give default value. if the variable or object is not null it returns its original value, if its null it returns the default value*/

println("\n'is' operator") /*'is' operator is used to check whether the object or variable is of particular type or not. same as instanceof in Java*/
var a:Any = 1
if(a is String){
println("Is a String")
} else{
println("Not a String")
}
}